### DISCUSSION TITLE
how to install ytdlp during using this project?

### DISCUSSION THREAD

**Discussion by UmbraCi** [1715173329/UnblockNeteaseMusic#408] [Q&A] [answered]
environment：win10 + node v14
i use this project by run command node app.js -p 52000:52001,then appear a CMD windows.
But output message :ERROR (YtDlpNotInstalled): (provider/yt-dlp) You must install "yt-dlp" before using the "ytdlp" source.
then i try to github download ytdlp.exe for Windows.
but run ytdlp.exe the dos for ytdlp.exe flash across.
and the Chinese on unlock server dos output is  messy code
![image](https://user-images.githubusercontent.com/49011112/147197609-e7be518a-9e68-47b1-b183-2ca3d8cecbea.png)

**Comment by pan93412** [comment, marked as answer]
You may need to rename `ytdlp.exe` to `yt-dlp.exe` and place it to one of the path listed in your environment variable `PATH`.

It would be much better to install it with `winget` or `choco` rather than place the executable file manually.

**Comment by UmbraCi** [comment]
> You may need to rename `ytdlp.exe` to `yt-dlp.exe` and place it to one of the path listed in your environment variable `PATH`.

how to resolve Chinese on unlock server dos output is messy code

**Comment by pan93412** [comment]
For the garbled text in your terminal, you may need to change your `cmd` encoding method to `UTF-8` or `Unicode`. That's an old issue in Windows.
